frontend/packages/text-editor/src/TextEntry.tsx

```diff
@@ -19,25 +22,41 @@ export const TextEntry = ({
 }: TextEntryProps) => {
   const [textEntryValue, setTextEntryValue] = useState(translation);
   const textareaRef = useAutoSizeTextArea(textEntryValue);
+  const { t } = useTranslation();
 
   const variables = [];
 
   const handleTextEntryChange = (e: React.ChangeEvent<HTMLTextAreaElement>) =>
     setTextEntryValue(e.currentTarget.value);
 
-  const handleTextEntryBlur = () =>
+  const handleTextEntryBlur = () => {
+    if (textEntryValue === '') return;
     upsertTextResource({ language: lang, translation: textEntryValue, textId });
+  };
 
   return (
     <div className={className}>
-      <LegacyTextArea
-        aria-label={lang + ' translation'}
-        value={textEntryValue}
-        onBlur={handleTextEntryBlur}
-        onChange={handleTextEntryChange}
-        ref={textareaRef}
-        resize={'vertical'}
-      />
+      <FormField
+        value={textId}
+        customValidationRules={() => {
+          if (textId === APP_NAME && textEntryValue === '') return 'TekstSouldNotBeEmpty';
+          return '';
+        }}
+        customValidationMessages={(errorCode: string) => {
+          if (errorCode === 'TekstSouldNotBeEmpty') return t('validation_errors.required');
+        }}
+        renderField={({ fieldProps }) => (
+          <Textarea
+            {...fieldProps}
+            aria-label={lang + ' translation'}
+            value={textEntryValue}
+            onBlur={handleTextEntryBlur}
+            onChange={handleTextEntryChange}
+            ref={textareaRef}
+            size='small'
+          />
```

Comment:
I am not sure about this, so this is more like a question, but can some of these props be included in `{...fieldProps}` such as `value` and `onChange` ? 🤔 Based on the default props in `fieldProps`:

```
export type FormFieldChildProps<TT> = {
  value: any;
  label: string;
  onChange: (value: TT, event?: React.ChangeEvent<HTMLInputElement>) => void;
  'aria-errormessage'?: string;
  'aria-invalid'?: boolean;
};
```

Reply:
Yes, but we explicitly defined them here for spesifc `value `and `onChange`

Reply:
Thx for the discussion we had on this one, that clarified things for me :)